specs/agents/tracing-instrumentation-grpc.md

```diff
@@ -28,10 +29,53 @@ See [apm#180](https://github.com/elastic/apm/issues/180) and [apm#115](https://g
 * **name**: \<method\>, ex: `/helloworld.Greeter/SayHello`
 * **type**: `external`
 * **subtype**: `grpc`
+* **outcome**: See [Outcome](#outcome)
 * **destination**:
   * **address**: Either an IP (v4 or v6) or a host/domain name.
   * **port**: A port number; Should report default ports.
   * **service**:
     * **resource**: Capture host, and port.
     * **name**: Capture the scheme, host, and non-default port.
     * **type**: Same as `span.type`
+
+#### Outcome
+
+With gRPC, transaction and span outcome is set from gRPC response status.
+
+If such status is not available, then we default to the following:
+
+- `failure` if an error is reported
+- `success` otherwise
+
+According to the [gRPC status codes reference spec](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md), some
+statuses are not used by gRPC client & server, thus they should be considered as client-side errors.
+
+The gRPC `UNKNOWN` status refers to an error that is not known, thus we should treat it as a `failure` and NOT map it to
+an `unknown` outcome.
+
+For gRPC spans (from the client):
+
+- `OK` : `success`
+- anything else: `failure`
+
+For gRPC transactions (from the server):
+
+- `OK` : `success`
+- `CANCELLED` : `failure`
+- `UNKNOWN` : `failure`
+- `INVALID_ARGUMENT` : `success` (*)
+- `DEADLINE_EXCEEDED` : `failure`
+- `NOT_FOUND` : `success` (*)
+- `ALREADY_EXISTS` : `success` (*)
+- `PERMISSION_DENIED` : `success` (*)
+- `RESOURCE_EXHAUSTED` : `failure`
+- `FAILED_PRECONDITION` : `success` (*)
+- `ABORTED` : `success` (*)
+- `OUT_OF_RANGE` : `success` (*)
+- `UNIMPLEMENTED` : `failure`
+- `INTERNAL` : `failure`
+- `UNAVAILABLE` : `failure`
+- `DATA_LOSS` : `success` (*)
```

Comment:
The spec says: `Unrecoverable data loss or corruption`. Isn't that a server-side error?

Reply:
Fixed in the last commit